server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanPropertyCacheTest.java

```diff
@@ -0,0 +1,204 @@
+        Assert.assertEquals(1, observer.createdCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 22));
+        Assert.assertEquals(1, observer.createdCounter.intValue());
+    }
+
+    @Test
+    public void testObservableUpdated() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.updatedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.updatedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 111));
+        Assert.assertEquals(1, observer.updatedCounter.intValue());
+    }
+
+    @Test
+    public void testObservableRemoved() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 111));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 222));
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
+    }
+
+    @Test
+    public void testObservableOnNonExistingEntry() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 33));
+        Assert.assertEquals(1, observer.createdCounter.intValue());
+        Assert.assertEquals(0, observer.updatedCounter.intValue());
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+    }
+
+    @Test
+    public void testRemoveObserver() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+
+        Property<Integer> property = new Property<>("id1", 2);
+        cache.store(null, property);
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 34));
+        Assert.assertEquals(1, observer.updatedCounter.intValue());
+
+        cache.removeObserver(null, "id1", observer);
+        cache.store(null, new Property<>("id1", 34));
```

Comment:
This wouldn't update the update counter anyway, as it has the same value. To be sure, best to change the value from 34 to something else - but apart from that the test is fine.